draft-ietf-lake-edhoc.md

```diff
@@ -579,25 +579,38 @@ In order to reduce round trips and the number of messages, or to simplify proces
 
 ~~~~~~~~~~~ CDDL
-ead = 1* (
+EAD_x = 1* ead
+
+ead = (
   ead_label : int,
-  ead_value : bstr,
+  ? ead_value : bstr,
 )
 ~~~~~~~~~~~
 
-A security application using external authorization data need to register a positive ead_label and the associated ead_value format for each EAD item it uses (see {{iana-ead}}), and describe processing and security considerations. Each application registers their own EAD items and defines associated operations. The application may define multiple uses of certain EAD items, e.g., the same EAD item may be used in different EDHOC messages with the same application.
+A security application using external authorization data needs to register a positive ead_label and optionally an associated ead_value format, for each EAD item it uses (see {{iana-ead}}). Each application registers its own EAD items and describes the associated processing and security considerations. The application may define multiple uses of certain EAD items, e.g., the same EAD item may be used in different EDHOC messages with the same application. Multiple occurrences of an EAD item in one EAD field may also be specified.
 
-An EAD item can be either critical or non-critical, determined by the sign of the ead_label in the transported EAD item included in the EDHOC message. Using the registered positive value indicates that the EAD item is non-critical. The corresponding negative value indicates that the EAD item is critical. ead_label = 0 MUST NOT be used.
+An EAD item can be either critical or non-critical, determined by the sign of the ead_label in the EAD item transported in the EDHOC message. A negative value indicates that the EAD item is critical and a non-negative value indicates that this EAD item is non-critical. The specification registering a new EAD label needs to describe under what conditions the EAD item is critical or non-critical, and thus whether the ead_label is used with negative or positive sign. ead_label = 0 is used for padding, see {{padding}}.
 
 If an endpoint receives a critical EAD item it does not recognize or a critical EAD item that contains information that it cannot process, the EDHOC protocol MUST be discontinued. A non-critical EAD item can be ignored.
 
-The specification registering a new EAD label needs to describe under what conditions the EAD item is critical or non-critical.
-
 The EAD fields of EDHOC must not be used for generic application data. Examples of the use of EAD are provided in {{ead-appendix}}.
 
+### Padding {#padding}
+
+EDHOC message_1 and the plaintext of message_2, message_3 and message_4 can be padded with the use of the corresponding EAD_x field, for x  = 1, 2, 3, 4. Padding is intended to be discarded by the receiving application.
+
+Padding is obtained by using an EAD item with ead_label = 0 and a random byte string of appropriate length as ead_value, noting that the ead_label and the CBOR encoding of ead_value also add bytes. For example:
```

Comment:
We are not saying anything about what requirements "random" implies here.
(Clearly, the recipient cannot test on randomness 😄 )

Reply:
Other properties of the byte string has been discussed. @emanjon

Reply:
No requirement on "random", but "random" is what we need for Christians use case where he wants hide information in the padding EAD without intermediaries seeing a difference.

Maybe we can write that the sender sets the value to a random string and the receiver ignores it.

Reply:
We already say that the sender sets the value to a random byte string and the receiver ignores it. 

The word "random" which always brings questions. Is "pseudo-random" better? Then it becomes an operational condition, use output from a PRNG; rather than a question whether the byte string is truly random. 

We should also formulate to avoid questions about short paddings. The empty byte string is not random.

Reply:
Proposal in 4f5b2c2